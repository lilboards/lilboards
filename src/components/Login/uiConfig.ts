import firebase from 'firebase';

/**
 * Configure FirebaseUI.
 *
 * @see {@link https://github.com/firebase/firebaseui-web-react}
 */
const uiConfig = {
  // Popup sign-in flow rather than redirect flow.
  signInFlow: 'popup',
  // Display auth providers.
  signInOptions: [
    firebase.auth.GoogleAuthProvider.PROVIDER_ID,
    firebase.auth.EmailAuthProvider.PROVIDER_ID,
  ],
  callbacks: {
    // Avoid redirects after sign-in.
    signInSuccessWithAuthResult: () => false,
  },
};

export default uiConfig;
