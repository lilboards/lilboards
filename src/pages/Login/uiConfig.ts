/**
 * Configure FirebaseUI.
 *
 * @see {@link https://github.com/firebase/firebaseui-web-react}
 */
const uiConfig = {
  // Popup sign-in flow rather than redirect flow.
  signInFlow: 'popup',
  // Display auth providers.
  signInOptions: [
    'google.com', // require('firebase/auth').GoogleAuthProvider.PROVIDER_ID
    'github.com', // require('firebase/auth').GithubAuthProvider.PROVIDER_ID
    'twitter.com', // require('firebase/auth').TwitterAuthProvider.PROVIDER_ID
    'password', // require('firebase/auth').EmailAuthProvider.PROVIDER_ID
  ],
  callbacks: {
    // Avoid redirects after sign-in.
    signInSuccessWithAuthResult: () => false,
  },
};

export default uiConfig;
