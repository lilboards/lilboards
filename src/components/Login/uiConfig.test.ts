import uiConfig from './uiConfig';

it('matches FirebaseUI config', () => {
  expect(uiConfig).toMatchSnapshot();
});

it('avoids redirect after sign-in', () => {
  expect(uiConfig.callbacks.signInSuccessWithAuthResult()).toBe(false);
});
