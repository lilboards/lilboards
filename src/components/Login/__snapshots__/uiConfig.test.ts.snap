// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches FirebaseUI config 1`] = `
Object {
  "callbacks": Object {
    "signInSuccessWithAuthResult": [Function],
  },
  "signInFlow": "popup",
  "signInOptions": Array [
    "google.com",
    "github.com",
    "twitter.com",
    "password",
  ],
}
`;
